src/compiler/core.rs

```diff
@@ -85,12 +86,27 @@ pub fn concatenate<TItem>(mut array1: Vec<TItem>, mut array2: Vec<TItem>) -> Vec
     array1
 }
 
-pub fn insert_sorted<TItem>(array: &mut SortedArray<TItem>, insert: TItem) {
+fn identity_key_selector<TItem>(item: TItem, _: Option<usize>) -> TItem {
```

Comment:
I can delete this, ended up not using it (was when I was seeing if using `fn`'s instead of `Fn*`'s (ie closures) would get rid of any lifetime issues)